src/prop/minisat/core/Solver.cc

```diff
@@ -934,13 +1058,30 @@ int Solver::analyze(CRef confl, vec<Lit>& out_learnt, int& out_btlevel)
         seen[var(p)] = 0;
         pathC--;
 
-        if (options::unsatCores() && pathC > 0 && confl != CRef_Undef)
+        if (pathC > 0 && confl != CRef_Undef)
         {
-          ProofManager::getSatProof()->addResolutionStep(p, confl, sign(p));
+          if (options::unsatCores())
+          {
+            ProofManager::getSatProof()->addResolutionStep(p, confl, sign(p));
+          }
+          if (isProofEnabled())
+          {
+            d_pfManager->addResolutionStep(ca[confl], p);
+          }
         }
 
     }while (pathC > 0);
     out_learnt[0] = ~p;
+    if (CVC4::options::proofNew())
```

Comment:
```suggestion
    if (options::proofNew())
```

Reply:
Should this just be guarded with `Debug.isOn("newproof::sat")`?

Reply:
Yes :)